sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubProducerClient.cs

```diff
@@ -59,88 +67,257 @@ public class EventHubProducer : IAsyncDisposable
+        /// <param name="connectionString">The connection string to use for connecting to the Event Hubs namespace; it is expected that the Event Hub name and SAS token are contained in this connection string.</param>
+        /// <param name="eventHubName">The name of the specific Event Hub to associate the connection with.</param>
+        /// <param name="producerOptions">A set of options to apply when configuring the producer.</param>
         ///
         /// <remarks>
-        ///   Because this is a non-public constructor, it is assumed that the <paramref name="producerOptions" /> passed are
-        ///   owned by this instance and are safe from changes made by consumers.  It is considered the responsibility of the
-        ///   caller to ensure that any needed cloning of options is performed.
+        ///   If the connection string is copied from the Event Hub itself, it will contain the name of the desired Event Hub,
+        ///   and can be used directly without passing the <paramref name="eventHubName" />.  The name of the Event Hub should be
+        ///   passed only once, either as part of the connection string or separately.
         /// </remarks>
         ///
-        internal EventHubProducer(TransportEventHubProducer transportProducer,
-                                  Uri endpoint,
-                                  string eventHubName,
-                                  EventHubProducerOptions producerOptions,
-                                  EventHubRetryPolicy retryPolicy)
+        public EventHubProducerClient(string connectionString,
+                                      string eventHubName,
+                                      EventHubProducerClientOptions producerOptions)
         {
-            Argument.AssertNotNull(transportProducer, nameof(transportProducer));
-            Argument.AssertNotNull(endpoint, nameof(endpoint));
+            Argument.AssertNotNullOrEmpty(connectionString, nameof(connectionString));
+            producerOptions = producerOptions?.Clone() ?? new EventHubProducerClientOptions();
+
+            OwnsConnection = true;
+            Connection = new EventHubConnection(connectionString, eventHubName, producerOptions.ConnectionOptions);
+            PartitionId = producerOptions.PartitionId;
+            RetryPolicy = producerOptions.RetryOptions.ToRetryPolicy();
+            InnerProducer = Connection.CreateTransportProducer(producerOptions);
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="EventHubProducerClient"/> class.
+        /// </summary>
+        ///
+        /// <param name="fullyQualifiedNamespace">The fully qualified Event Hubs namespace to connect to.  This is likely to be similar to <c>{yournamespace}.servicebus.windows.net</c>.</param>
+        /// <param name="eventHubName">The name of the specific Event Hub to associated the connection with.</param>
```

Comment:
typo: `associate`

Reply:
It might be a good idea to run an all-solution search on the `to associated the` expression. This might have been generated by a large-scale copy-paste.